src/kagglehub/http_resolver.py

```diff
@@ -320,8 +313,11 @@ def _build_get_instance_url_path(h: ModelHandle) -> str:
     return f"models/{h.owner}/{h.model}/{h.framework}/{h.variation}/get"
 
 
-def _build_download_url_path(h: ModelHandle) -> str:
-    return f"models/{h.owner}/{h.model}/{h.framework}/{h.variation}/{h.version}/download"
+def _build_download_url_path(h: ModelHandle, path: Optional[str]) -> str:
```

Comment:
nit: Ack pre-existing but please rename to `_build_model_download_url_path` to match other cases.